subprojects/core/src/main/java/org/gradle/api/internal/tasks/userinput/ChoiceBuilder.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.userinput;
+
+import java.util.function.Function;
+
+public interface ChoiceBuilder<T> {
```

Comment:
🤔 This type has 2 implementations, both called `Default`.  Could one be called `Interactive` and one `NonInteractive?

Reply:
Done.